docs/reference/troubleshooting/discovery-issues.asciidoc

```diff
@@ -72,21 +83,38 @@ When a node wins the master election, it logs a message containing
 `elected-as-master`. If this happens repeatedly, the elected master node is
 unstable. In this situation, focus on the logs from the master-eligible nodes
 to understand why the election winner stops being the master and triggers
-another election.
+another election. If the logs suggest that the master is unstable due to
+timeouts or network-related issues then narrow down the problem as follows.
+
+include::network-timeouts.asciidoc[tag=troubleshooting-network-timeouts-gc-vm]
+
+include::network-timeouts.asciidoc[tag=troubleshooting-network-timeouts-packet-capture-elections]
+
+include::network-timeouts.asciidoc[tag=troubleshooting-network-timeouts-threads]
 
 [discrete]
 [[discovery-cannot-join-master]]
 === Node cannot discover or join stable master
 
 If there is a stable elected master but a node can't discover or join its
 cluster, it will repeatedly log messages about the problem using the
-`ClusterFormationFailureHelper` logger. Other log messages on the affected node
-and the elected master may provide additional information about the problem.
+`ClusterFormationFailureHelper` logger. The <<health-api>> API on the affected
+node will also provide useful information about the situation. Other log
+messages on the affected node and the elected master may provide additional
+information about the problem. If the logs suggest that the node cannot
+discover or join the cluster due to timeouts or network-related issues then
+narrow down the problem as follows.
+
+include::network-timeouts.asciidoc[tag=troubleshooting-network-timeouts-gc-vm]
```

Comment:
We have basically 3 copies of the same long list of bullet points, in the same documentation page, very close to each other. I wonder if maybe refactoring the list into a section in the same page, and referencing that section from these 3 places, might be a bit better?

Reply:
Typically there's only one section which matches the situation in which a cluster finds itself, and folks are (empirically) terrible at reading contextual information or following links, so I would like to keep the duplication for the sake of clarity instead of refactoring it out as you suggest.